### ISSUE TITLE
Improve CLI documentation

### ISSUE THREAD

**Issue by bisubus** [hestiacp/hestiacp#1271]
I've been having problems with comprehending CLI functionality from [existing documentation](https://docs.hestiacp.com/cli_commands.html) and I'm sure I'm not the only one. There's some learning curve for daily use, docs don't help much for occasional use when you don't know what exactly you look for.

There are categories. They break alphabetic order and make it harder to search by keywords (there are no descriptions to look in anyway) while I'm not sure that the separation is justified. *DNS* and *Mail* seem accurate, although I found `v-list-remote-dns-hosts` in *Common*. But what is the reasoning for *Common* and *Panel*? Is *Panel* for functionality that is backed by the panel or the functionality for Hestia management? If it's the former then it would be better to be a label rather than a category, and if it's the latter then it's wrong because it deals with things that aren't specific to CP, e.g. firewall.

The documentation is stale, I compared it against current command list and the difference is like 20%. Descriptions are missing, they are present in https://vestacp.com/docs/cli/ that's also in alphabetic order but there's still a considerable difference between Vesta and Hestia commands.

Formatting is not a big deal when you get used to it but still less clean than monospaced uppercased/lowercased params. Option names are capitalized and read as a song name:
> Parameters: User Archive Directory

Descriptions are available in Vesta docs, so they are in scripts but they are scarce and aren't always as descriptive as they could be. That Hestia docs have examples that Vesta docs lack provides some context but still requires to check sources almost every time to make sure that a script does what I want it to do. Longer descriptions that explain how a command works and affected by params would be very welcome.

To sum it up, I see some space for improvements:

- Keep commands in alphabetic order and replace categories with labels for the ease of browsing, they could be filtered with a widget like it was done in [Angular API](https://angular.io/api)
- Add `hestia` label to designate commands that are unique to it for those who migrate from Vesta
- Provide descriptions in a TOC because command names are ambiguous, could be link `title` to save a space
- Generate documentation from script comments (they currently lack examples and category/label sections)
- More detailed descriptions

**Comment by jaapmarcus** [comment]
I sadly have to agree. But it is an terrible job to update / add missing items to the list

**Comment by bisubus** [comment]
That's the reason why generating docs from scratch seems like a good idea, comment block in scripts looks very consistent for parsing but it likely needs to be updated with missing data (examples and tags). I'll likely end up doing this myself for reference but it's unlikely this will be a format and tools that are acceptable for PR.

**Comment by ScIT-Raphael** [comment]
@bisubus it would be more than awesome if you could rearrange your code for a pr, this would help to bring the project forward <3!

**Comment by jaapmarcus** [comment]
Documentation is based on based on [Sphinx](https://www.sphinx-doc.org/en/master/) and can be found on https://github.com/hestiacp/hestiacp-docs

It is possible to run it on you own server or locally if needed... 

There is currently an script available that generate the documentation how ever this order all command on one page alphabetical. 

https://github.com/hestiacp/hestiacp/blob/main/src/hst_generate_clidocs.sh

**Comment by bisubus** [comment]
Thanks for the reference. I ended up merging examples from those docs. I wouldn't dare to approach this in shell so tools are made with Node, https://github.com/bisubus/hestia-cli-docs . This fixes all formatting issues and most inaccuracies so far.  I submitted a PR that will allow to easily use annotation comment from scripts as single source of truth. Considering that the repo doesn't use other languages for tools, it could be not a great decision to bring Node or Python to it but at this point comment parsing will be a breeze, potentially in shell too, although I'll leave it to Bash ninjas.

Any way, the tool creates JSON output that can be used already. Here is usable [markdown doc](https://github.com/bisubus/hestia-cli-docs/blob/master/docs/README.md) that was generated from it.

**Comment by jaapmarcus** [comment]
When running the commnand

v-add-dns-domain: possible options mismatch, 9/12
v-add-dns-record: description punctuation missing
v-add-fs-archive: possible options mismatch, 4/7
v-add-user-composer: description punctuation missing
v-change-sys-config-value: description punctuation missing
v-change-sys-release: description punctuation missing
v-get-sys-timezones: possible options mismatch, 1/2
v-list-sys-services: possible options mismatch, 1/2
v-search-command: missing # on empty line
v-search-command: possible options mismatch, 2/0
v-search-command: extra comments
v-update-sys-hestia-git: missing # on empty line
v-update-sys-hestia-git: extra comments

**Comment by bisubus** [comment]
@jaapmarcus It does some heuristics at possible problems. IIRC these are false positives and this is expected output ATM.

`missing # on empty line` and `extra comments` preemptively check for missing #, since all of this was fixed in #1278 this can be removed from the tool to reduce the noise.  `description punctuation missing` in [v-add-user-composer](https://github.com/hestiacp/hestiacp/blob/main/bin/v-add-user-composer) refers to second description line that doesn't read well in a sentence when a description is unwrapped, another one is false positive. `possible options mismatch` is fuzzy by nature.

**Comment by jaapmarcus** [comment]
I know it more as reference what need to be checked pull request looks nice it has been merged.

Modified your code slightly so it outputs .rst instead of .md it works fine

**Comment by jaapmarcus** [comment]
Updated https://docs.hestiacp.com 

Thanks for your contribution :heart:
